src/pdm/cli/commands/list.py

```diff
@@ -1,37 +1,369 @@
 import argparse
+import json
+from collections import defaultdict
+from typing import DefaultDict, Dict, List, Optional, Sequence, Set
 
 from pdm.cli import actions
 from pdm.cli.commands.base import BaseCommand
+from pdm.cli.utils import (
+    build_dependency_graph,
+    check_project_file,
+    get_dist_location,
+    show_dependency_graph,
+)
+from pdm.compat import importlib_metadata as im
+from pdm.exceptions import PdmUsageError
 from pdm.project import Project
 
+# Group label for subdependencies
+SUBDEP_GROUP_LABEL = ":sub"
+
 
 class Command(BaseCommand):
     """List packages installed in the current working set"""
 
     def add_arguments(self, parser: argparse.ArgumentParser) -> None:
-        group = parser.add_mutually_exclusive_group()
-        group.add_argument(
+        graph = parser.add_mutually_exclusive_group()
```

Comment:
You seem to leave the `--freeze` option out, that isn't what I mean in the matrix. `--freeze` is supposed work without any other options. `pdm list --freeze` prints the **currently installed** requirements list compatible with `pip`'s requirements file format.

Reply:
Ah sorry, I thought you meant to remove it.  I will add it back 👍

Reply:
Sorted 👍 will push asap